src/java.net.http/share/classes/jdk/internal/net/http/Http1Response.java

```diff
@@ -479,11 +366,12 @@ public void onComplete() {
                 });
```

Comment:
> Here we don't need to check whether it's been acquired because we know it's been acquired if we reach here.

Hello Daniel, may be I am misreading the diff but from what I can see, the `acquire()` method now returns a `boolean` which we are assigning to the local `acquire`. But we aren't checking it for `true` for doing the subsequent operation. So when it reaches the finally block here, as far as I can see, it's not guaranteed that we did indeed `acquire` it.

Reply:
The acquire() method will return true the first time it's been called. And it is called only once. So we only need to check whether `acquired` is true at places where we are in doubt about whether the method is called.
The code that calls acquire() above could simply have ignored the result of `acquire()` and set the boolean to `true`. That said, it would not be wrong to check whether `acquired==true` here too - maybe I should do it for consistency...

Reply:
> The acquire() method will return true the first time it's been called. And it is called only once. So we only need to check whether acquired is true at places where we are in doubt about whether the method has been called.

I see what you mean.

Reply:
Instead of checking `acquired`  in the lambda - which requires `acquired` being final, I will add a line to assert that `assert acquired == true` after calling `acquire()`.